src/components/Documents/DocumentCard.jsx

```diff
@@ -0,0 +1,231 @@
+// React Imports
+import React, { useState } from 'react';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import FileOpenIcon from '@mui/icons-material/FileOpen';
+import ShareIcon from '@mui/icons-material/Share';
+import DeleteOutlineOutlinedIcon from '@mui/icons-material/DeleteOutlineOutlined';
+import IconButton from '@mui/material/IconButton';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import Typography from '@mui/material/Typography';
+
+// Utility Imports
+import { truncateText } from '@utils';
+
+/**
+ * @typedef {object} Document
+ * @property {string} name - The given name of the document
+ * @property {string} type - The given type of the document
+ * @property {string} description - The given description of the document
+ * @property {string} uploadDate- The upload date of the document
+ * @property {string} endDate - The expiration date of the document
+ * @property {string} fileUrl - The file URL of the document
+ */
+
+/**
+ * DocumentCard - Component that contains a document
+ *
+ * @memberof Documents
+ * @name DocumentCard
+ * @param {object} Props - Component props for Document Preview
+ * @param {Document} Props.document - The document
+ * @param {Function} Props.onPreview - The document preview event
+ * @param {Function} Props.onShare - The document share event
+ * @param {Function} Props.onDelete - The document delete event
+ * @returns {React.JSX.Element} React component for DocumentCard
+ */
+const DocumentCard = ({ document, onShare, onDelete, onPreview }) => {
+  // primary columns
+  // name, type, uploaddate, expiration, actions
+  // expandable information
+  //  description
+
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [openMenu, setOpenMenu] = useState(null);
+
+  const handleClick = (event, clickedDocument) => {
+    setAnchorEl(event.currentTarget);
+    setOpenMenu(clickedDocument.id);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+    setOpenMenu(null);
+  };
+
+  const handleMenuItemClick = (action, clickedDocument) => () => {
+    action(clickedDocument);
+    handleClose();
+  };
+
+  const getTypeText = (type) => {
+    switch (type) {
+      case 'driversLicense':
+        return "Driver's License";
+      case 'passport':
+        return 'Passport';
+      case 'bankStatement':
+        return 'Bank Statement';
+      default:
```

Comment:
I would like to see a case for 'Other' documents since we have that as a specific option. What I'm not sure about is whether to add a separate case for it or to change this default.